### DISCUSSION TITLE
Where and how overlapping blocks can come from?

### DISCUSSION THREAD

**Discussion by Tusenka** [prometheus/prometheus#15001] [Q&A]
https://github.com/prometheus/prometheus/blob/main/tsdb/compact.go#L258

**Comment by bboreham** [comment]
Some commentary here: https://ganeshvernekar.com/blog/prometheus-tsdb-compaction-and-retention/#condition-1-overlapping-blocks

Also I believe out-of-order data can generate overlapping blocks (which is new since Ganesh's blog was written).
